android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/dao/HivRegisterDao.kt

```diff
@@ -101,6 +103,38 @@ constructor(
       .filterNot { it.healthStatus == HealthStatus.DEFAULT }
   }
 
+  override suspend fun searchByName(
+    nameQuery: String,
+    currentPage: Int,
+    appFeatureName: String?
+  ): List<RegisterData> {
+    val patients =
+      fhirEngine.search<Patient> {
+        filter(
+          Patient.NAME,
+          {
+            modifier = StringFilterModifier.CONTAINS
+            value = nameQuery
+          }
+        )
+        filter(Patient.ACTIVE, { value = of(true) })
+        filter(Patient.GENDER, { value = of(CodeType("male")) }, { value = of(CodeType("female")) })
+        sort(Patient.NAME, Order.ASCENDING)
+        count = PaginationConstant.DEFAULT_PAGE_SIZE
+        from = currentPage * PaginationConstant.DEFAULT_PAGE_SIZE
+      }
```

Comment:
Seems like this does not take into filtration through the `identifier` and ART number as previously done [here](https://github.com/opensrp/fhircore/blob/7d8644c9dd1dee066a67d7a55a375f77987e3f21/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/register/PatientRegisterViewModel.kt#L182)